### PR TITLE
[Doc] Add Customization of Dataset in doc

### PR DESCRIPTION
**Pull request by MengzhangLI**
## Motivation

Make documentation of dataset customization richer, which makes it eaiser for beginers.

### CODE REVIEW

File: docs/en/tutorials/customize_datasets.md [hunk 1]
```diff
@@ -1,5 +1,181 @@
...
+After your dataset conversion done above, create a new file `my_dataset.py` in `./mmseg/dataset/`, where defining `MyDataset` class could be registered into `DATASETS` of MMCV and be used by model.
+
+
+```python
+from .builder import DATASETS
+from .custom import CustomDataset
+
+# Register MyDataset class into DATASETS
+@DATASETS.register_module()
+class MyDataset(CustomDataset):
```
Comment (MeowZheng) [inline docs/en/tutorials/customize_datasets.md:65]: Might explain the main interfaces and their function of CustomDataset

File: docs/en/tutorials/customize_datasets.md [hunk 2]
```diff
@@ -1,5 +1,181 @@
...
+            img_suffix='.png',
+            seg_map_suffix='.png',
+            reduce_zero_label=False, # reduce_zero_label is False because label 0 is background (first one in CLASSES above)
+            **kwargs)
+```
+
+`CLASSES` and `PALETTE` are defined in `./mmseg/dataset/my_dataset.py`, which are class names and BGR values of annotations, respectively.
+Besides,`PALETTE` would only be used in prediction visulization, which would not affect process of training and validation.
+
+Specifically, if label 0 in segmentation map is not background, it should be set `reduce_zero_label=True`.
```
Comment (MeowZheng) [inline docs/en/tutorials/customize_datasets.md:85]: might provide more clear explanation for `reduce_zero_label`

File: docs/en/tutorials/customize_datasets.md (hunk 1 above)
Comment (MengzhangLI) [reply docs/en/tutorials/customize_datasets.md:65]: > 
In my opinion, it is better to help them run the code, as for explanations of advanced class of parameters like `CustomDataset` and `reduce_zero_label`, I wonder if it pushes too much strength on beginners who are totally new to this codebase.

**Comment by SidhartheneeNayak** [comment]
Hello everyone, I am trying to fine to the "segmenter_vit-s_linear_8x1_512x512_160k_ade20k" model on another small-sized dataset. But whenever I am trying to train the model, it is showing an error stating that it expects a 4d input but gets a 3d input. I know the 4th dimension is the batch size but I am not getting exactly where to change the batch_size in the configuration file or how to modify my input 2d RGB images.
Below is the code for configuration: 


from mmcv import Config
cfg = Config.fromfile('configs/segmenter/segmenter_vit-s_mask_8x1_512x512_160k_ade20k.py')

from mmseg.apis import set_random_seed

# Since we use only one GPU, BN is used instead of SyncBN
cfg.norm_cfg = dict(type='BN', requires_grad=True)
cfg.model.backbone.norm_cfg = cfg.norm_cfg
#######cfg.model.backbone.mlp_ratio = 8 ####################
cfg.model.decode_head.norm_cfg = cfg.norm_cfg
#cfg.model.auxiliary_head.norm_cfg = cfg.norm_cfg

# modify num classes of the model in decode/auxiliary head
cfg.model.decode_head.num_classes = 8
#cfg.model.auxiliary_head.num_classes = 8

# Modify dataset type and path
cfg.dataset_type = 'StanfordBackgroundDataset_rescaled'
cfg.data_root = data_root

cfg.data.samples_per_gpu = 1
cfg.data.workers_per_gpu=1

cfg.img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
cfg.crop_size = (256, 256)
cfg.train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=cfg.crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **cfg.img_norm_cfg),
    dict(type='Pad', size=cfg.crop_size, pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
]

cfg.test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 512),
        # img_ratios=[0.5, 0.75, 1.0, 1.25, 1.5, 1.75],
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **cfg.img_norm_cfg),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]


cfg.data.train.type = cfg.dataset_type
cfg.data.train.data_root = cfg.data_root
cfg.data.train.img_dir = img_dir
cfg.data.train.ann_dir = ann_dir
cfg.data.train.pipeline = cfg.train_pipeline
cfg.data.train.split = 'splits/train.txt'

cfg.data.val.type = cfg.dataset_type
cfg.data.val.data_root = cfg.data_root
cfg.data.val.img_dir = img_dir
cfg.data.val.ann_dir = ann_dir
cfg.data.val.pipeline = cfg.test_pipeline
cfg.data.val.split = 'splits/val.txt'

cfg.data.test.type = cfg.dataset_type
cfg.data.test.data_root = cfg.data_root
cfg.data.test.img_dir = img_dir
cfg.data.test.ann_dir = ann_dir
cfg.data.test.pipeline = cfg.test_pipeline
cfg.data.test.split = 'splits/val.txt'

# We can still use the pre-trained model though we do not need to
# use the mask branch
cfg.load_from = 'checkpoints/segmenter_vit-s_mask_8x1_512x512_160k_ade20k_20220105_151706-511bb103.pth'

# Set up working dir to save files and logs.
cfg.work_dir = './work_dirs/tutorial'

cfg.runner.max_iters = 200
cfg.log_config.interval = 10
cfg.evaluation.interval = 200
cfg.checkpoint_config.interval = 200

# Set seed to facitate reproducing the result
cfg.seed = 0
set_random_seed(0, deterministic=False)
cfg.gpu_ids = range(1)
cfg.device = 'cuda' ############################

# Let's have a look at the final config used for training
print(f'Config:\n{cfg.pretty_text}')



Below is the code for training the new dataset which is showing the error:


from mmseg.datasets import build_dataset
from mmseg.models import build_segmentor
from mmseg.apis import train_segmentor


# Build the dataset
datasets = [build_dataset(cfg.data.train)]

# Build the detector
model = build_segmentor(cfg.model)
# Add an attribute for visualization convenience
model.CLASSES = datasets[0].CLASSES

# Create work_dir
mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))
train_segmentor(model, datasets, cfg, distributed=False, validate=True, 
                meta=dict())



Below I have atttached the screenshot of the error:
Screenshot 2022-06-22 182604
